Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs

```diff
@@ -0,0 +1,583 @@
+                                newRot = new Vector3(newPos.x, 0.0f, newPos.z);
+                                NodeList[i].transform.rotation = Quaternion.LookRotation(newRot);
+                                break;
+
+                            case OrientTypeEnum.FaceOriginReversed:
+                                newRot = new Vector3(newPos.x, 0f, newPos.z);
+                                NodeList[i].transform.rotation = Quaternion.LookRotation(newRot);
+                                NodeList[i].transform.Rotate(0f, 180f, 0f);
+                                break;
+
+                            case OrientTypeEnum.FaceFoward:
+                                NodeList[i].transform.forward = transform.forward;
+                                break;
+
+                            case OrientTypeEnum.FaceForwardReversed:
+                                NodeList[i].transform.forward = transform.forward;
+                                NodeList[i].transform.Rotate(0f, 180f, 0f);
+                                break;
+                        }
+                        NodeList[i].transform.localPosition = newPos;
+                    }
+                    break;
+                case SurfaceTypeEnum.Sphere:
+
+                    for (int i = 0; i < NodeList.Count; i++)
+                    {
+                        newPos = SphericalMapping(nodeGrid[i], Radius);
+                        switch (OrientType)
+                        {
+                            case OrientTypeEnum.None:
+                            default:
+                                // Don't apply any rotation
+                                break;
+
+                            case OrientTypeEnum.FaceOrigin:
+                                newRot = newPos;
+                                NodeList[i].transform.rotation = Quaternion.LookRotation(newRot);
+                                break;
+
+                            case OrientTypeEnum.FaceOriginReversed:
+                                newRot = newPos;
+                                NodeList[i].transform.rotation = Quaternion.LookRotation(newRot);
+                                NodeList[i].transform.Rotate(0f, 180f, 0f);
+                                break;
+
+                            case OrientTypeEnum.FaceFoward:
+                                NodeList[i].transform.forward = transform.forward;
+                                break;
+
+                            case OrientTypeEnum.FaceForwardReversed:
+                                NodeList[i].transform.forward = transform.forward;
+                                NodeList[i].transform.Rotate(0f, 180f, 0f);
+                                break;
+                        }
+                        NodeList[i].transform.localPosition = newPos;
+                    }
+                    break;
+
+                case SurfaceTypeEnum.Scatter:
+					// TODO fix this rough first commit! Magic numbers galore!
```

Comment:
We should probably address this TODO

Reply:
I believe that's an old comment that is no longer valid as there aren't Magic numbers galore.